### PR TITLE
[PluggableDevice] add int32 DEVICE_DEFAULT for BroadcastTo/Pack/StridedSlice

### PR DESCRIPTION
**Pull request by quintinwang5**
Add int32 DEVICE_DEFAULT for BroadcastTo/Pack/StridedSlice, this PR is for PluggableDevice.

### CODE REVIEW

**Comment by quintinwang5** [comment]
Hi @penpornk , There are also 3 PRs for `int32` `DEVICE_DEFAULT`. Thanks.
```
https://github.com/tensorflow/tensorflow/pull/51777
https://github.com/tensorflow/tensorflow/pull/51762
https://github.com/tensorflow/tensorflow/pull/51761
```

**Comment by gbaned** [comment]
@penpornk  Can you please review this PR ? Thanks!

**Comment by gbaned** [comment]
@penpornk Can you please review this PR ? Thanks!

**Comment by gbaned** [comment]
@penpornk Can you please review this PR ? Thanks!

File: tensorflow/core/kernels/broadcast_to_op.cc [hunk 1]
```diff
@@ -130,16 +130,13 @@ TF_CALL_GPU_ALL_TYPES(REGISTER_KERNEL);
 TF_CALL_int64(REGISTER_KERNEL);
 #undef REGISTER_KERNEL
 
-// A special GPU kernel for int32.
-// TODO(b/25387198): Also enable int32 in device memory. This kernel
-// registration requires all int32 inputs and outputs to be in host memory.
```
Comment (penpornk) [inline tensorflow/core/kernels/broadcast_to_op.cc:135]: Please add back these comment lines and change GPU to DEVICE_DEFAULT.

File: tensorflow/core/kernels/pack_op.cc [hunk 2]
```diff
@@ -151,16 +151,13 @@ TF_CALL_uint64(REGISTER_GPU);
 TF_CALL_GPU_ALL_TYPES(REGISTER_GPU);
 #undef REGISTER_GPU
 
-// A special GPU kernel for int32.
-// TODO(b/25387198): Also enable int32 in device memory. This kernel
-// registration requires all int32 inputs and outputs to be in host memory.
```
Comment (penpornk) [inline tensorflow/core/kernels/pack_op.cc:156]: Please add back these comment lines and change GPU to DEVICE_DEFAULT.

**Review by penpornk — APPROVED** [review]
Thank you for the PR and I'm sorry for the delay!

**Review by rthadur — APPROVED** [review]

**Review by penpornk — APPROVED** [review]

**Pull request merged**